src/components/LayoutTree.tsx

```diff
@@ -890,6 +1041,26 @@ const LayoutTree = () => {
     treeThoughts,
   ])
 
+  // Calculate isGreaterThought at LayoutTree level
+  const cursorIsGreaterThanPrevSibling = useSelector(state => {
+    try {
+      if (!state.cursor) return false
+
+      const cursorThought = getThoughtById(state, head(state.cursor))
+      if (!cursorThought) return false
+
+      // Get the current thought and previous thought to compare
+      const index = treeThoughtsPositioned.findIndex(thought => thought.isCursor)
+      const prevThought = index > 0 ? getThoughtById(state, head(treeThoughtsPositioned[index - 1].path)) : null
+
+      // Compare the current thought with the previous thought to determine the animation curve
+      const thoughtComparison: number = prevThought && cursorThought ? compareThought(cursorThought, prevThought) : 0
+      return thoughtComparison > 0
+    } catch (error) {
+      return false
+    }
```

Comment:
try-catch is generally only for unhandleable runtime errors. Instead, detect the appropriate condition and return false in that case.

Note that try-catch also comes with a performance cost, so that's another reason not to use it except in the case of unexpected runtime faults.